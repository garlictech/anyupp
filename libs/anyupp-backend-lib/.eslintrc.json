{
  "extends": "../../.eslintrc.json",
  "ignorePatterns": [
    "!**/*"
  ],
  "rules": {},
  "overrides": [
    {
      "files": [
        "*.ts"
      ],
      "plugins": [
        "@typescript-eslint"
      ],
      "parserOptions": {
        "project": [
          "libs/anyupp-backend-lib/tsconfig.*?.json"
        ]
      }
    }
  ]
}
