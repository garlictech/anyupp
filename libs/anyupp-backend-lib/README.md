# anyupp-backend-lib

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test anyupp-backend-lib` to execute the unit tests via [Jest](https://jestjs.io).
