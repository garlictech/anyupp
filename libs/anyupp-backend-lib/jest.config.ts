module.exports = {
  displayName: 'anyupp-backend-lib',

  globals: {
    'ts-jest': { tsconfig: '<rootDir>/tsconfig.spec.json' },
  },
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/anyupp-backend-lib',
  preset: '../../jest.preset.ts',
};
