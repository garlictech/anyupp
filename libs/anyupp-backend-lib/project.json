{
  "root": "libs/anyupp-backend-lib",
  "sourceRoot": "libs/anyupp-backend-lib/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/anyupp-backend-lib/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/anyupp-backend-lib"],
      "options": {
        "jestConfig": "libs/anyupp-backend-lib/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": ["platform:nodejs", "type:backend"]
}
