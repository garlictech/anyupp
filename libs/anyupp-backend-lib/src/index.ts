export * from './lib/report';
export * from './lib/seeder';
export * from './lib/infra';
export * from './lib/database';
export * from './lib/date/date.utils';
export * from './lib/resolver-utils/get-authenticated-userid';
