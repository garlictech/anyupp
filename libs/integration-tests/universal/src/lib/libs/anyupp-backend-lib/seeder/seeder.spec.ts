import { seedAdminUser, seedBusinessData } from '@bgap/anyupp-backend-lib';
import {
  createTestChain,
  createTestChainProduct,
} from '@bgap/anyupp-backend-lib';

describe('Seeder test', () => {
  it.skip('should createTestChain work', done => {
    createTestChain(1).subscribe({
      next(result: any) {
        expect(result).toHaveProperty('createChain');
        const {
          createdAt,
          updatedAt,
          ...justCreatedObject
        } = result.createChain;
        expect(createdAt).not.toBeUndefined();
        expect(updatedAt).not.toBeUndefined();
        expect(justCreatedObject).toMatchSnapshot();
        done();
      },
    });
  }, 25000);

  it.skip('should createTestChainProduct work', done => {
    createTestChainProduct(1, 1, 1).subscribe({
      next(result: any) {
        expect(result).toHaveProperty('createChainProduct');
        const {
          createdAt,
          updatedAt,
          ...justCreatedObject
        } = result.createChainProduct;
        expect(createdAt).not.toBeUndefined();
        expect(updatedAt).not.toBeUndefined();
        expect(justCreatedObject).toMatchSnapshot();
        done();
      },
    });
  }, 25000);

  it.skip('should the seedBusinessData function run successfully', done => {
    seedBusinessData().subscribe({
      next() {
        done();
      },
    });
  }, 25000);
});
