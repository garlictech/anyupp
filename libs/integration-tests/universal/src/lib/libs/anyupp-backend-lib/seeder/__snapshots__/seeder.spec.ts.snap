// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seeder test should createTestChain work 1`] = `
Object {
  "__typename": "Chain",
  "address": null,
  "description": Object {
    "__typename": "LocalizedItem",
    "de": null,
    "en": "Test chain #1 description",
    "hu": "Teszt lánc #1 leírás",
  },
  "email": "info@chain1.com",
  "id": "chain_1_id",
  "isActive": true,
  "name": "Test chain #1",
  "phone": "1234567890",
  "style": Object {
    "__typename": "ChainStyle",
    "colors": Object {
      "__typename": "ChainStyleColors",
      "backgroundDark": "#fff",
      "backgroundLight": "#fff",
      "borderDark": "#fff",
      "borderLight": "#fff",
      "disabled": "#fff",
      "highlight": "#fff",
      "indicator": "#fff",
      "textDark": "#fff",
      "textLight": "#fff",
    },
    "images": null,
  },
}
`;

exports[`Seeder test should createTestChainProduct work 1`] = `
Object {
  "__typename": "ChainProduct",
  "chainId": "chain_1_id",
  "description": Object {
    "__typename": "LocalizedItem",
    "de": null,
    "en": "Test chain termék #1 description",
    "hu": "Teszt chain termék #1 leírás",
  },
  "id": "chain_product_c1_1_id",
  "image": null,
  "isVisible": true,
  "name": Object {
    "__typename": "LocalizedItem",
    "de": null,
    "en": "Test chain product #1 name",
    "hu": "Teszt chain termék #1 név",
  },
  "productCategoryId": "product_category_c1_1_id",
  "productType": "drink",
  "variants": Array [
    Object {
      "__typename": "ProductVariant",
      "availabilities": null,
      "id": "chain_product_variant_c1_p1_1_id",
      "isAvailable": true,
      "pack": Object {
        "__typename": "ProductVariantPack",
        "size": 2,
        "unit": "dl",
      },
      "position": 10,
      "price": 11,
      "refGroupPrice": null,
      "variantName": Object {
        "__typename": "LocalizedItem",
        "de": null,
        "en": "glass",
        "hu": "pohár",
      },
    },
  ],
}
`;
